circuits/kimchi/src/expr.rs

```diff
@@ -1047,10 +1048,10 @@ impl<F: Field> Expr<ConstantExpr<F>> {
 
     /// Combines multiple constraints `[c0, ..., cn]` into a single constraint
     /// `alpha^alpha0 * c0 + alpha^{alpha0 + 1} * c1 + ... + alpha^{alpha0 + n} * cn`.
-    pub fn combine_constraints(alpha0: usize, cs: Vec<Self>) -> Self {
+    pub fn combine_constraints(alphas: &mut impl Iterator<Item = usize>, cs: Vec<Self>) -> Self {
         let zero = Expr::<ConstantExpr<F>>::zero();
         cs.into_iter()
-            .zip(alpha0..)
+            .zip_eq(alphas)
```

Comment:
this is to make sure that we don't create LESS constraints if we set `alphas` to a range that's too small. (This will panic if we do this).